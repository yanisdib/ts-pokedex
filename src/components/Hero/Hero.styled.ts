import styled, { css } from 'styled-components';


const heading = css`
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    height: 50%;
`;

export const Section = styled.section`
    display: flex;
    position: relative;
    flex-direction: row;
    width: 100%;
    height: 94vh;
    margin-bottom: 50px;
`;

export const Wrapper = styled.div`
    display: flex;
    flex-direction: column;
    width: inherit;
    height: 100%;
`;

export const Title = styled.div`
    ${heading};

    h1 {
        @media (min-width: 375px) {
            width: 100%;
            font-size: 5em;
            font-weight: 800;
            text-align: center;
            text-transform: uppercase;
            margin: 20px 0px;
        }

        @media (min-width: 950px) {
            font-size: 8em;
            margin: 30px 100px;
        }

        @media (min-width: 1128px) {
            text-align: end;
        }

        @media (min-width: 1814px) {
            text-align: end;
        }
    }
`;

export const Branding = styled.div`
    ${heading};

    align-items: flex-end;

    svg {
        height: 30px;
        fill: var(--ui-content-color);
    }

    h2 {
        @media (min-width: 375px) {
            width: 100%;
            font-size: 1em;
            font-weight: 700;
            text-align: center;
            text-transform: uppercase;
            margin: 20px 0;
        }

        @media (min-width: 950px) {
            width: 100%;
            font-size: 1.2em;
            text-align: end;
            margin: 20px 100px;
        }

        @media (min-width: 1128px) {
            width: 15%;
        }

        @media (min-width: 1814px) {
            width: 10%;
        }
    }
`;


interface StartButtonProps {
    onClick?: () => void;
}


export const StartButton = styled.button<StartButtonProps>`
    display: flex;
    align-self: center;
    justify-content: center;
    width: 250px;
    height: fit-content;
    background-color: transparent;
    font-family: var(--pixel-font);
    font-size: 1em;
    color: var(--ui-content-color);
    text-transform: uppercase;
    animation: blinker 1.3s linear infinite;
`;