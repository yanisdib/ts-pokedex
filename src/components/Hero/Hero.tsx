import { useContext } from 'react';

import { DarkModeContext } from '../../contexts/DarkModeContext';

import { playConfirmSound } from '../../helpers/utils/sounds';

import { ThemeSwitch } from '..';

import {
    Branding,
    Section,
    StartButton,
    Title,
    Wrapper
} from './Hero.styled';


function Hero(): JSX.Element {
    const setDarkMode = useContext(DarkModeContext);

    return (
        <Section>
            <Wrapper>
                <Title>
                    <h1>Pokémon</h1>
                </Title>
                <StartButton onClick={playConfirmSound}>
                    Press start
                </StartButton>
                <Branding>
                    <ThemeSwitch onClick={setDarkMode} />
                    <h2>ポケットモンスター Pokédex ® 1995</h2>
                </Branding>
            </Wrapper>
        </Section>
    );
}


export default Hero;