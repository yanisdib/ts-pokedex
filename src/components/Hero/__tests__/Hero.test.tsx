import { render } from '@testing-library/react';

import Hero from '../Hero';


describe('Hero component', () => {
    test('should render without crashing', () => {
        const { container } = render(<Hero />);

        expect(container).toMatchSnapshot();
    });
});