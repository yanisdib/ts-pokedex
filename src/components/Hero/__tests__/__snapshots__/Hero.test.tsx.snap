// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component should render without crashing 1`] = `
<div>
  <section
    class="sc-iBkjds fSmdMU"
  >
    <div
      class="sc-ftvSup AwCYN"
    >
      <div
        class="sc-papXJ eILoLV"
      >
        <h1>
          Pokémon
        </h1>
      </div>
      <button
        class="sc-kDDrLX jDhDEP"
      >
        Press start
      </button>
      <div
        class="sc-jqUVSM fYlZOm"
      >
        <span
          class="sc-gKXOVf iRSPxX"
        />
        <h2>
          ポケットモンスター Pokédex ® 1995
        </h2>
      </div>
    </div>
  </section>
</div>
`;
